### PR TITLE
[IMP] payment_*: set password widget on sensible fields

### PR DESCRIPTION
**Pull request by hote-odoo**
When setting up an acquirer there is sensible information to
be supplied.
There was inconsistency on what was obfuscated and what not.
Now sensible information as passwords and keys is obfuscated
and public information as names and addresses is visible.

task-2694139

### CODE REVIEW

**Comment by chevalierv** [comment]
LGTM :smiley: 

Tip: You should change your commit title to better target the modules you change (e.g.: _payment(\_*)_ )

File: addons/payment_mollie/views/payment_views.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
         <field name="arch" type="xml">
             <xpath expr='//group[@name="acquirer"]' position='inside'>
                 <group attrs="{'invisible': [('provider', '!=', 'mollie')]}">
-                    <field name="mollie_api_key" string="API Key" attrs="{'required': [('provider', '=', 'mollie'), ('state', '!=', 'disabled')]}" password="True"/>
+                    <field name="mollie_api_key" password="True" string="API Key" attrs="{'required': [('provider', '=', 'mollie'), ('state', '!=', 'disabled')]}"/>
```
Comment (AntoineVDV) [inline addons/payment_mollie/views/payment_views.xml:10]: Why this change? The `password` attribute is not essential to the comprehension of the view and should therefore stay at the end of the list of attributes.

Comment (AntoineVDV) [reply addons/payment_mollie/views/payment_views.xml:10]: Same thing for other fields.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Please change the commit title to `[IMP] payment_*: set password widget on sensible fields` to better follow the guidelines and avoid implying that `payment` is impacted.

File: addons/payment_mollie/views/payment_views.xml (hunk 1 above)
Comment (hote-odoo) [reply addons/payment_mollie/views/payment_views.xml]: done :)  (both remarks)

File: addons/payment_payumoney/views/payment_views.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
         <field name="arch" type="xml">
             <xpath expr='//group[@name="acquirer"]' position='inside'>
                 <group attrs="{'invisible': [('provider', '!=', 'payumoney')]}">
-                    <field name="payumoney_merchant_key" attrs="{'required':[ ('provider', '=', 'payumoney'), ('state', '!=', 'disabled')]}"/>
+                    <field name="payumoney_merchant_key" attrs="{'required':[ ('provider', '=', 'payumoney'), ('state', '!=', 'disabled')]}" password="True"/>
```
Comment (AntoineVDV) [inline addons/payment_payumoney/views/payment_views.xml:11]: public

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
The following fields are public and the password widget should therefore be removed for them:
- `alipay_merchant_partner_id`
- `authorize_client_key`
- `stripe_publishable_key`

Could you please also remove the `groups` attribute on `authorize_currency_id`? The field is not private.

File: addons/payment_adyen/views/payment_views.xml [hunk 3]
```diff
@@ -9,9 +9,9 @@
             <xpath expr='//group[@name="acquirer"]' position='inside'>
                 <group attrs="{'invisible': [('provider', '!=', 'adyen')]}">
                     <field name="adyen_merchant_account" attrs="{'required':[('provider', '=', 'adyen'), ('state', '!=', 'disabled')]}"/>
-                    <field name="adyen_api_key" attrs="{'required':[('provider', '=', 'adyen'), ('state', '!=', 'disabled')]}"/>
-                    <field name="adyen_client_key" attrs="{'required':[('provider', '=', 'adyen'), ('state', '!=', 'disabled')]}"/>
-                    <field name="adyen_hmac_key" attrs="{'required':[('provider', '=', 'adyen'), ('state', '!=', 'disabled')]}"/>
+                    <field name="adyen_api_key" attrs="{'required':[('provider', '=', 'adyen'), ('state', '!=', 'disabled')]}" password="True"/>
+                    <field name="adyen_client_key" attrs="{'required':[('provider', '=', 'adyen'), ('state', '!=', 'disabled')]}" password="True"/>
```
[the remark this answers is not in the document]
Comment (hote-odoo) [reply addons/payment_adyen/views/payment_views.xml]: done :)

File: addons/payment_buckaroo/views/payment_views.xml [hunk 4]
```diff
@@ -8,8 +8,8 @@
         <field name="arch" type="xml">
             <xpath expr='//group[@name="acquirer"]' position='inside'>
                 <group attrs="{'invisible': [('provider', '!=', 'buckaroo')]}">
-                    <field name="buckaroo_website_key" attrs="{'required':[ ('provider', '=', 'buckaroo'), ('state', '!=', 'disabled')]}"/>
-                    <field name="buckaroo_secret_key" string="Secret Key" attrs="{'required':[ ('provider', '=', 'buckaroo'), ('state', '!=', 'disabled')]}"/>
+                    <field name="buckaroo_website_key" attrs="{'required':[ ('provider', '=', 'buckaroo'), ('state', '!=', 'disabled')]}" password="True"/>
```
[the remark this answers is not in the document]
Comment (hote-odoo) [reply addons/payment_buckaroo/views/payment_views.xml]: done :)

File: addons/payment_payulatam/views/payment_views.xml [hunk 5]
```diff
@@ -11,9 +11,11 @@
                     <field name="payulatam_merchant_id"
                            attrs="{'required':[('provider', '=', 'payulatam'), ('state', '!=', 'disabled')]}"/>
                     <field name="payulatam_account_id"
-                           attrs="{'required':[('provider', '=', 'payulatam'), ('state', '!=', 'disabled')]}"/>
+                           attrs="{'required':[('provider', '=', 'payulatam'), ('state', '!=', 'disabled')]}"
+                           password="True"/>
```
[the remark this answers is not in the document]
Comment (hote-odoo) [reply addons/payment_payulatam/views/payment_views.xml]: done :)

File: addons/payment_payumoney/views/payment_views.xml (hunk 2 above)
Comment (hote-odoo) [reply addons/payment_payumoney/views/payment_views.xml]: done :)

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
The changes requested in https://github.com/odoo/odoo/pull/81211#pullrequestreview-893850120 are still not applied.

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+
